plugins/core-plugin/src/main/resources/Dockerfile-template-yaml

```diff
@@ -17,18 +17,24 @@ WORKDIR $WORKDIR
 # Throw if requirements.txt file not provided
 RUN if ! [ -f requirements.txt ] ; then >&2 echo "error: no requirements.txt file found" && exit 1 ; fi
 
+# Set up custom PyPi repository, if applicable
+${airlockConfig}
+
 # Install dependencies to launch the pipeline and download to reduce startup time
-# TODO - remove `pip uninstall apache-beam` line when repo is upgraded to Beam 2.59.0
 RUN python -m venv /venv \
-    && /venv/bin/pip uninstall apache-beam -y \
-    && /venv/bin/pip install --require-hashes --no-cache-dir -U -r $REQUIREMENTS_FILE \
+    && /venv/bin/pip install --require-hashes --no-deps --no-cache-dir -U -r $REQUIREMENTS_FILE \
     && /venv/bin/pip download --require-hashes --no-cache-dir --dest /tmp/dataflow-requirements-cache -r $REQUIREMENTS_FILE \
-    && rm -rf /usr/local/lib/python$PY_VERSION/site-packages  \
+    && rm -rf /usr/local/lib/python$PY_VERSION/site-packages \
     && mv /venv/lib/python$PY_VERSION/site-packages /usr/local/lib/python$PY_VERSION/
 
 # Cache provider environments for faster startup and expansion time
 RUN python -m apache_beam.yaml.cache_provider_artifacts
 
+<#if saSecretName?? && airlockJavaRepo??>
+COPY cache_provider_jars.py /template
+RUN mkdir -p $HOME/.apache_beam/cache/jars
+RUN python cache_provider_jars.py --cache_dir="/root/.apache_beam/cache/jars/" --maven_central_repository_url="https://_json_key_base64:$(gcloud secrets versions access latest --secret=${saSecretName} | base64 -w 0)@us-maven.pkg.dev/artifact-foundry-prod/${airlockJavaRepo}"
+</#if>
```

Comment:
What is our goal with this block? Would be good to have a comment explaining what we are doing here